java/ql/src/semmle/code/java/security/XPath.qll

```diff
@@ -0,0 +1,58 @@
+/** Provides classes to reason about XPath vulnerabilities. */
+
+import java
+import semmle.code.java.dataflow.DataFlow
+import semmle.code.java.dataflow.ExternalFlow
```

Comment:
The `ExternalFlow` import needs to be bidirectional. I.e. add a `private import semmle.code.java.security.XPath` inside the `private module Frameworks` in `ExternalFlow.qll`. This is important to ensure that `ExternalFlow.qll` always sees the same set of csv rows to enable efficient caching. Otherwise, a lot of things will be recomputed based on which csv rows are transitively imported.

Reply:
Fixed in b69be30. Another "gotcha" to keep in mind, thanks!

Reply:
I really want to make this a compilation error, but we'll need a little bit of additional language support for that.